src/pages/LeadTeamsPage.tsx

```diff
@@ -0,0 +1,64 @@
+import React, { useState, useEffect } from 'react';
+import { SimpleGrid, Heading, Spinner, Text, VStack, Input, InputGroup, InputLeftElement } from '@chakra-ui/react';
+import { SearchIcon } from '@chakra-ui/icons';
+import { TeamCard } from '../components/TeamCard';
+import { teamService } from '../service/teamService';
+import { ITeam } from '../interfaces/ITeam';
+import { Layout } from '../components/Layout';
+
+export const LeadTeamsPage: React.FC = () => {
+  const [teams, setTeams] = useState<ITeam[]>([]);
+  const [filteredTeams, setFilteredTeams] = useState<ITeam[]>([]);
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState<string | null>(null);
```

Comment:
nit: instead of string | null you can use implicit undefined like so:

useState<string>()

looks cleaner imo, and only working with undefined as opposed to nulls and undefined is probably simpler

(i only just learned this recently so i'm sure there are contradictions to this all over the codebase rn)